internal/data/common.go

```diff
@@ -126,7 +128,161 @@ func (a AttributeValue) SetBool(v bool) {
 	a.orig.BoolValue = v
 }
 
+// AttributeKeyValue stores a key and AttributeValue pair.
+type AttributeKeyValue struct {
+	orig *otlpcommon.AttributeKeyValue
+}
+
+// NewAttributeKeyValue creates a new AttributeKeyValue with the given key.
+func NewAttributeKeyValue(k string) AttributeKeyValue {
+	return AttributeKeyValue{&otlpcommon.AttributeKeyValue{Key: k}}
+}
+
+// NewAttributeKeyValueString creates a new AttributeKeyValue with the given key and string value.
+func NewAttributeKeyValueString(k string, v string) AttributeKeyValue {
+	akv := AttributeKeyValue{&otlpcommon.AttributeKeyValue{Key: k}}
+	akv.Value().SetString(v)
+	return akv
+}
+
+// NewAttributeKeyValueInt creates a new AttributeKeyValue with the given key and int64 value.
+func NewAttributeKeyValueInt(k string, v int64) AttributeKeyValue {
+	akv := AttributeKeyValue{&otlpcommon.AttributeKeyValue{Key: k}}
+	akv.Value().SetInt(v)
+	return akv
+}
+
+// NewAttributeKeyValueDouble creates a new AttributeKeyValue with the given key and float64 value.
+func NewAttributeKeyValueDouble(k string, v float64) AttributeKeyValue {
+	akv := AttributeKeyValue{&otlpcommon.AttributeKeyValue{Key: k}}
+	akv.Value().SetDouble(v)
+	return akv
+}
+
+// NewAttributeKeyValueBool creates a new AttributeKeyValue with the given key and bool value.
+func NewAttributeKeyValueBool(k string, v bool) AttributeKeyValue {
+	akv := AttributeKeyValue{&otlpcommon.AttributeKeyValue{Key: k}}
+	akv.Value().SetBool(v)
+	return akv
+}
+
+// Key returns the key associated with this AttributeKeyValue.
+func (akv AttributeKeyValue) Key() string {
+	return akv.orig.Key
+}
+
+// Value returns the value associated with this AttributeKeyValue.
+func (akv AttributeKeyValue) Value() AttributeValue {
+	return AttributeValue{akv.orig}
+}
+
+// SetValue replaces the value associated with this AttributeKeyValue.
+func (akv AttributeKeyValue) SetValue(av AttributeValue) {
+	akv.orig.Type = av.orig.Type
+	akv.orig.StringValue = av.orig.StringValue
+	akv.orig.IntValue = av.orig.IntValue
+	akv.orig.DoubleValue = av.orig.DoubleValue
+	akv.orig.BoolValue = av.orig.BoolValue
+}
+
+// AttributeMap stores a map of attribute keys to values.
+type AttributeMap struct {
```

Comment:
The name looks a bit confusing. I see it does implement an interface similar to map, but did you consider any other name options?

Reply:
I did not consider other options, I like the current name (personal opinion). Do you have any suggestions (open to change if you have a better suggestion)?

Reply:
`AttributeSliceMap`? :)

Reply:
I'd suggest to keep the name. The fact that it uses a slice is an implementation detail and may even change in the future.
Semantics and the set of operations this type implements are of a classic Map.

Reply:
I was thinking of something like just `Attributes`, but it's good enough. Let's keep `AttributesMap`